Jetpack blocks: Remove 💥

sirreal opened the pull request:
Remove the `jetpack-blocks` package.

## Testing

Calypso isn't broken 🙌 
Any references to the package lying around to be cleaned up?

sirreal committed:
Remove jetpack-blocks package

sirreal committed:
Remove babel override

ockham committed:
Remove from CODEOWNERS

ockham commented:
Wondering how soon we can actually merge this. Do we need it around until the Jetpack 7.2 release in case we need to fix a block (or would we do that from within the JP repo :scream:)? Or even longer so we can use it for a point release, if necessary? :slightly_frowning_face:

ockham committed:
Remove now-obsolete mapbox-gl dep

simison commented:
This folder was there to use by calypso components/sections and blocks:

https://github.com/Automattic/wp-calypso/tree/deadf7401e0f5ada6f7f647b62b52f211349a005/client/assets

...those image files can prolly be moved back to calypso components and folder removed.

simison commented:
>Wondering how soon we can actually merge this. Do we need it around until the Jetpack 7.2 release in case we need to fix a block (or would we do that from within the JP repo 😱)? Or even longer so we can use it for a point release, if necessary? 🙁

We have a branch where we can pull changes from Jetpack and do a release for 7.2 from there: https://github.com/Automattic/wp-calypso/tree/update/jetpack-blocks-72

sirreal commented:
> Do we need it around until the Jetpack 7.2

I think we can move forward with this any time and let #31793 be the only home for blocks (bugfixes and be ported and released from there) until the `jetpack-blocks` package is deprecated.

sirreal committed:
Remove unused copy-webpack-plugin

sirreal commented on packages/jetpack-blocks/.eslintrc.js:
```diff
@@ -1,23 +0,0 @@
-module.exports = {
```
Should we move this into `client/gutenberg/extensions`? It aligns with the babel override that's in place there.

simison replied:
That's a good idea.

scruffian replied:
I had to remove `import/no-extraneous-dependencies`, as there's no `package.json` in this dir.

simison replied:
done in https://github.com/Automattic/wp-calypso/pull/31759/commits/a5b305ed2171cae85165a284fa56e1a456c3dabd

simison replied:
Looks like these configs were also in overrides file:

https://github.com/Automattic/wp-calypso/blob/8f4f95a70f5327175c54dddb2fe7b8057ee0593e/.eslintrc.js#L24-L46

...so I removed the config file in https://github.com/Automattic/wp-calypso/pull/31759/commits/eca830cae495502d6e7ed082c55732f567acad50

scruffian committed:
Move the .eslintrc to the gutenberg extensions dir

scruffian commented:
> This folder was there to use by calypso components/sections and blocks:
> 
> https://github.com/Automattic/wp-calypso/tree/deadf7401e0f5ada6f7f647b62b52f211349a005/client/assets
> 
> ...those image files can prolly be moved back to calypso components and folder removed.

@simison I don't see these files in Jetpack, so I think we should probably open another PR to move them there first?

ockham commented:
CircleCI's `build-notifications` task is failing, since it can't find `mini-css-extract-plugin-with-rtl` (and I can repro locally). This is kinda weird, the plugin is a dependency of `@automattic/calypso-build`, which in turn is a dependency of our root `package.json` :thinking:

ockham commented:
> > This folder was there to use by calypso components/sections and blocks:
> > https://github.com/Automattic/wp-calypso/tree/deadf7401e0f5ada6f7f647b62b52f211349a005/client/assets
> > ...those image files can prolly be moved back to calypso components and folder removed.
> 
> @simison I don't see these files in Jetpack, so I think we should probably open another PR to move them there first?

No, that's the point -- Jetpack doesn't need those. They were used by the `related-posts` block (example content) for a while, but now they aren't anymore. They're only used by Calypso, so we can (And should) move them back to the `assets/` dir, as Mikael is suggesting.

scruffian commented:
> No, that's the point -- Jetpack doesn't need those. They were used by the `related-posts` block (example content) for a while, but now they aren't anymore. They're only used by Calypso, so we can (And should) move them back to the `assets/` dir, as Mikael is suggesting.

Ok, on it :)

ockham committed:
Shrinkwrap

ockham commented:
> > No, that's the point -- Jetpack doesn't need those. They were used by the `related-posts` block (example content) for a while, but now they aren't anymore. They're only used by Calypso, so we can (And should) move them back to the `assets/` dir, as Mikael is suggesting.
> 
> Ok, on it :)

Per Slack convo, we decided to defer to a follow-up PR.

simison committed:
Remove Jetpack blocks mention in client/gutenberg/extensions/README.md

simison reviewed (approved):
This is looking good now. 👍 

We're moving ⛰ s here!

simison committed:
Remove extensions specific eslint file as dublicate

...these are already in https://github.com/Automattic/wp-calypso/blob/8f4f95a70f5327175c54dddb2fe7b8057ee0593e/.eslintrc.js#L24-L46

scruffian commented:
> Per Slack convo, we decided to defer to a follow-up PR.

Done in https://github.com/Automattic/wp-calypso/pull/31800
